src/main/webapp/pagination.js

```diff
@@ -9,9 +9,152 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+let fileLength = Number.MAX_VALUE;
+
+// the function contains algo for fetching page 
+// and updating current page's content
 async function changePage(page) {
-    // TODO :- the function contains algo for fetching data 
-    // and updating current page's content
-    // WILL BE ADDED ALONG WITH PAGINATION
+    // change current page (page user is viewing at present) to the 
+    // value provided while calling the function 
+    updateCurrentPage(page);
+    const logs = document.getElementById(LOGS).getAttribute("aria-selected");
+    const searchString = document.getElementById(SEARCH_BAR).value;
+    const fileType = logs == "true" ? LOGS : ERRORS ;
+    // get the page no to be fetched next 
+    const fetchPage = getPageToBeFetched();
+    let fileName = document.getElementById(FILE_NAME).value;
+    // remove blank spaces from fileName
+    fileName =  fileName.trim();
+
+    // if we are on page 1 we need to reset the value of last page 
+    // else we already have requested page in window which we can simply
+    // display 
+    if(page == 1){
+        await resetLastPage(fileName, fileType);
+    } else {
+        display();
+    }
+
+    // if fetched page != -1 that means we need to fetch data either to create window(page == 1)
+    // or to maintain window(page > 1)
+    if(fetchPage != -1) {
+        // get file offset for page to be fetched  
+        const pageSpecs = getPageStartAndSize(fileType, fetchPage, searchString);
+        // fetch data from database
+        fetchedData = await callPaginationServlet(fileName, fileType, searchString, pageSpecs);
+        // update value of last page
+        updateLastPageBasedOnFetchedPageLength(fetchedData.length, fetchPage);
+        // if the length of first page is zero that means
+        //  there is no file with given name in our database
+        if(fetchedData.length == 0 && currentPage == 1) {
+            fileNotFound();
+            return;
+        }
+        // add fetched page to window we are maintaining
+        addToData(fetchedData, fetchPage, fileType);    
+    }
+
+    // if currentPage == 1 then we have just created window 
+    // and now we have requested page with us in window and 
+    // we can show it to user
+    if(currentPage == 1 ) {
+        display();
+    }
+}
+
+// reset value of lastPage when on page 1
+async function resetLastPage(fileName, fileType) {
+    updateLastPage(Number.MAX_VALUE);
+    updateNoOfRecordsOnLastPage(recordsPerPage)
+
+    // if we are dealing with error file we want to 
+    // show it in bottom up manner (error ranking)
+    // for that reason we can't provide last page a 
+    // default value because we will be needing it to 
+    // calculate file offsets
+    if(fileType == ERRORS){
+        // get length of file by making a servlet call
+        fileLength = await getCount(fileName, fileType);
+        updateLastPage(Math.ceil(fileLength/recordsPerPage));
+        if(fileLength % recordsPerPage != 0)
```

Comment:
What if the value is equal to 0?
We should update it to recordsPerPage right?

Reply:
we already did that in line-69 while setting default value now we just need to check of != 0  we need to override default value